Pre SDK power update (Android)

stephanie-anderson opened the pull request:
- move files from `/common` to base dir, since no guides exist
- moved integrations from `/configuration/integrations` to `/integrations` as preparation for the SDK power update
  - moved all integrations in their own subfolder `/integrations/<integration>` in order to improve the overview of all integrations (reduce noise by potential images used on integration pages)
  - added redirects for all integrations

romtsn commented:
I believe we should also add redirects for those pages that are moved:
```
redirect_from:
  - /platforms/android/configuration/integrations/<integration_name>/
```

stephanie-anderson commented:
> I believe we should also add redirects for those pages that are moved, to not break external links:
> 
> ```
> redirect_from:
>   - /platforms/android/configuration/integrations/<integration_name>/
> ```

Thank you @romtsn! I added redirects as suggested :) I also moved the integrations in separate subfolders, to reduce the noise from images in the same folder

<img width="295" alt="Screenshot 2023-10-13 at 14 02 42" src="https://github.com/getsentry/sentry-docs/assets/7096858/c3c6a6af-b797-4b2a-bec9-32fa6e069017">

<br>

<img width="355" alt="Screenshot 2023-10-13 at 14 03 06" src="https://github.com/getsentry/sentry-docs/assets/7096858/10c16f52-3e4a-4493-9767-5062428afbb8">

shanamatthews reviewed (approved):
Looks good, thanks for adding the redirects!
